### PR TITLE
[CXF-6987] Also consider the classes defined on an application when …

### PR DESCRIPTION
**Pull request by johnament**
…singletons are populated.

### CODE REVIEW

**Comment by johnament** [comment]
Apologies, as some formatting changes went in as well.

File: integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java [hunk 1]
```diff
@@ -179,7 +185,13 @@ private JAXRSServerFactoryBean createFactoryInstance(final Application applicati
                 classified.get(Path.class).add(singleton);
             }
         }
-        
+
+        // now loop through the classes
+        Set<Class<?>> classes = application.getClasses();
+        if (!classes.isEmpty()) {
+            classified.get(Path.class).addAll(loadServices(beanManager, classes));
+            classified.get(Provider.class).addAll(loadProviders(beanManager, classes));
```
Comment (reta) [inline integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java]: Would you mind to add Feature.class as well? Yeah, it is CXF, not JAX-RS feature, but it won't hurt to have the same behavior as with singletons. Minor comment though, leaving choice to you. Thank you.

**Comment by reta** [comment]
LGTM, thank you. Good one, covers more JAX-RS/CDI scenarios.

File: integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java (hunk 1 above)
Comment (johnament) [inline integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java]: Shoot, thanks.  I actually wrote the code for features, but then removed it in my formatting clean up.

Comment (johnament) [inline integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java]: Fixed, thanks @reta

Comment (reta) [inline integration/cdi/src/main/java/org/apache/cxf/cdi/JAXRSCdiResourceExtension.java]: :+1: Thanks!

**Comment by johnament** [comment]
Restarted jenkins, test failure seems unrelated.

**Comment by reta** [comment]
Jenkins retest this please

**Comment by johnament** [comment]
Jenkins retest this please

**Comment by johnament** [comment]
I'm going to close and create a new PR to see if that will force jenkins to rebuild.  Would it be OK if I modified the job to have a longer history?

**Comment by johnament** [comment]
Oh, you merged it! Heh, thanks.

**Comment by sberyozkin** [comment]
Hi, re "Would it be OK if I modified the job to have a longer history?" - sounds reasonable, Dan, what do you think ?

**Pull request closed without merging**